api/metrics/v1/http.go

```diff
@@ -250,6 +254,10 @@ func NewHandler(read, write, rulesEndpoint *url.URL, upstreamCA []byte, opts ...
 				t.TLSClientConfig.RootCAs.AppendCertsFromPEM(upstreamCA)
 			}
 
+			if upstreamCert != nil {
+				t.TLSClientConfig.Certificates = append(t.TLSClientConfig.Certificates, *upstreamCert)
```

Comment:
`TLSClientConfig` can be `nil`. I also noticed that this seems to be the only handler where `t` is initialized using `http.DefaultTransport` instead of creating a new one, not part of this PR though.